test/framework/src/main/java/org/elasticsearch/test/AbstractXContentTestCase.java

```diff
@@ -48,16 +49,16 @@
             Supplier<T> instanceSupplier,
             CheckedBiConsumer<T, XContentBuilder, IOException> toXContent,
             CheckedFunction<XContentParser, T, IOException> fromXContent) {
-        return new XContentTester<T>(
-                createParser,
-                instanceSupplier,
-                (testInstance, xContentType) -> {
-                    try (XContentBuilder builder = XContentBuilder.builder(xContentType.xContent())) {
-                        toXContent.accept(testInstance, builder);
-                        return BytesReference.bytes(builder);
-                    }
-                },
-                fromXContent);
+        return new XContentTester<>(
+            createParser,
+            x -> instanceSupplier.get(),
```

Comment:
I see! Usually you don't need an xcontentType before you make the instance, but for search hits it is different!